chained_filter_controller/test/test_chained_filter.cpp

```diff
@@ -0,0 +1,93 @@
+#include <gtest/gtest.h>
+#include <memory>
+#include "rclcpp/rclcpp.hpp"
+#include "controller_interface/controller_interface.hpp"
+#include "chained_filter_controller/chained_filter.hpp"
+
+using chained_filter_controller::ChainedFilter;
+using controller_interface::CallbackReturn;
+
+class ChainedFilterControllerTest : public ::testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    controller_ = std::make_shared<ChainedFilter>();
+  }
+
+  std::shared_ptr<ChainedFilter> controller_;
+};
+
+TEST_F(ChainedFilterControllerTest, InitReturnsSuccess)
+{
+  auto result = controller_->init("chained_filter", "test_ns", 3, "", rclcpp::NodeOptions());
+  EXPECT_EQ(result, controller_interface::return_type::OK);
+}
+
+TEST_F(ChainedFilterControllerTest, ConfigureReturnsFailureWithNoParams)
+{
+  // Initialize the controller (should set up internal node)
+  ASSERT_EQ(
+    controller_->init(
+      "chained_filter",         // controller name
+      "test_ns",                // namespace
+      0,                        // intra-process (0 = disabled)
+      "test_node",              // RCL node name
+      rclcpp::NodeOptions()),   // node options
```

Comment:
where do you have this from? This is completely wrong, see 
https://github.com/ros-controls/ros2_control/blob/c19997a8e614773cf69e5f3e6ffd7b511f2a4f5c/controller_interface/include/controller_interface/controller_interface_base.hpp#L154-L156